fx.go

```diff
@@ -23,11 +30,40 @@ func init() {
 	}
 }
 
+func checkForUpdate() {
+	releaseUrl := "https://api.github.com/repos/metrue/fx/releases/latest"
+	resp, err := http.Get(releaseUrl)
+	if err != nil {
+		log.Debugf("Failed to fetch Github release page, error %v", err)
+		return
+	}
+	defer resp.Body.Close()
+
+	decoder := json.NewDecoder(resp.Body)
+	var releaseJSON struct {
+		Tag string `json:"tag_name"`
+		Url string `json:"html_url"`
+	}
+	if err := decoder.Decode(&releaseJSON); err != nil {
+		log.Debugf("Failed to decode Github release page JSON, error %v", err)
+		return
+	}
+	if matched, err := regexp.MatchString(`^(\d+\.)(\d+\.)(\d+)$`, releaseJSON.Tag); err != nil || !matched {
+		log.Debugf("Unofficial release %s?", releaseJSON.Tag)
+		return
+	}
+	log.Debugf("Latest release tag is %s", releaseJSON.Tag)
+	if releaseJSON.Tag != Version {
```

Comment:
Any reason we don't do the comparison with `greater ` or `>` here?

Reply:
Because comparison doesn't apply to literal version strings, e.g:

* `"0.7.3" > "0.7.1"` is true
* `"0.7.3" > "0.7.10"` is also true

To support semantic version comparison, maybe we need to introduce an additional sem-version package? so here for simplicity, I just use the equal operator to do the check.

Reply:
I see, your solution is enough for now, thanks.